app/javascript/article-form/articleForm.jsx

```diff
@@ -160,7 +187,41 @@ export class ArticleForm extends Component {
     }
   };
 
+  lintMarkdown = () => {
+    const options = {
+      ...LINT_OPTIONS,
+      strings: {
+        content: this.state.bodyMarkdown,
+      },
+    };
+    const { content: markdownLintErrors } = window.markdownlint.sync(options);
+    this.setState({ markdownLintErrors });
+  };
+
+  fetchMarkdownLint = async () => {
+    if (!window.markdownlint) {
+      const markdownItScript = document.createElement('script');
+      markdownItScript.setAttribute('src', '/assets/markdown-it.min.js');
+      document.body.appendChild(markdownItScript);
+
+      // The markdownlint script needs the first script to have finished loading first
+      markdownItScript.addEventListener('load', () => {
+        const markdownLintScript = document.createElement('script');
+        markdownLintScript.setAttribute(
+          'src',
+          '/assets/markdownlint-browser.min.js',
+        );
+        document.body.appendChild(markdownLintScript);
+
+        markdownLintScript.addEventListener('load', this.lintMarkdown);
+      });
+    } else {
+      this.lintMarkdown();
+    }
```

Comment:
The markdownlint lib is [designed to be used in the browser by adding as script tags](https://github.com/DavidAnson/markdownlint#browser) rather than importing from a module. @nickytonline if you can suggest a better way of loading it in - let me know 🙏 

This loads it in once asynchronously (when the Preview tab is clicked), and from that point on the `markdownlint` method is available on the window object. This whole method is called async so that linting the markdown doesn't block any other preview functionality.

Reply:
I installed `markdown-it` with `yarn add markdown-it` and I found `markdown-it.min.js` in the `node_modules/markdown-it/dist` folder so I think we can require that directly from there in the editor. It's the only dependency `markdownlint` declares so it'll be there :)  

`markdownlint` instead doesn't seem to offer a bundled browser version :( maybe we can put it in a `app/javascript/vendor` so we know it's a third party script.

It's not a small library but I don't think we have an alternative: https://bundlephobia.com/result?p=markdownlint@0.23.1